src/envs/behavior.py

```diff
@@ -301,6 +312,44 @@ def render(self,
     def _get_task_relevant_objects(self) -> List["ArticulatedObject"]:
         return list(self.igibson_behavior_env.task.object_scope.values())
 
+    def set_igibson_behavior_env(self, seed: int) -> None:
+        """Sets/resets the igibson_behavior_env."""
+        np.random.seed(seed)
+        self.igibson_behavior_env = behavior_env.BehaviorEnv(
+            config_file=self._config_file,
+            mode=CFG.behavior_mode,
+            action_timestep=CFG.behavior_action_timestep,
+            physics_timestep=CFG.behavior_physics_timestep,
+            action_filter="mobile_manipulation",
+            rng=self._rng,
+        )
+        self.igibson_behavior_env.step(
+            np.zeros(self.igibson_behavior_env.action_space.shape))
+        if CFG.behavior_randomize_init_state:
+            # NOTE: in this case, since we're attempting to randomize
+            # the initial state, iGibson might fail. If this happens,
+            # then at least one bddl object scope will be None. We keep
+            # trying to get a new environment until we succeed.
+            ig_objs_bddl_scope = [
+                self._ig_object_name(obj)
+                for obj in self._get_task_relevant_objects()
+            ]
+            while None in ig_objs_bddl_scope:
+                self.igibson_behavior_env = behavior_env.BehaviorEnv(
+                    config_file=self._config_file,
+                    mode=CFG.behavior_mode,
+                    action_timestep=CFG.behavior_action_timestep,
+                    physics_timestep=CFG.behavior_physics_timestep,
+                    action_filter="mobile_manipulation",
+                    rng=self._rng,
+                )
+                self.igibson_behavior_env.step(
+                    np.zeros(self.igibson_behavior_env.action_space.shape))
+                ig_objs_bddl_scope = [
+                    self._ig_object_name(obj)
+                    for obj in self._get_task_relevant_objects()
+                ]
```

Comment:
No need to duplicate the code in this loop with the code above. A better pattern is:
```
while True:
    self.igibson_behavior_env = blah
    self.igibson_behavior_env.step()
    ig_objs_bddl_scope = [blah]
    if None not in ig_objs_bddl_scope:
        break
```

Reply:
One concern is, how likely is it that this loop might run forever? Should we impose a max number of iterations before killing it or something?

Reply:
from our tests, there's a very low probability this will run forever, but I'll add a counter just in case!